ynput/ayon-usd
# Refactor Settings and fix launch environment

## Changelog Description


Refactor settings to match with feedback in #31 


## Additional info


~~This is a first pass draft, and likely does not work. Just wanted to put it up so there's visibility on that I have a first pass on this.~~ This now works for me, with both Houdini 20 and Maya 2024.2

- Bugfix: The used settings are now from the active bundle (and use `ayon_core` studio settings and project settings retrieval)
- Bugfix: Launch context environment is now maintained and used (this contains #32)
- Bugfix: Setup resolver env logic now does not 'kill' `PATH` on Windows by accident because it previously stacked to `LD_LIBRARY_PATH` and got set to `PATH`. It is now 'appended'` to `PATH`.
- Simplifies settings (and also changes the settings keys! So note that you need to reconfigure your settings)
- Improves logging for invalid server repository, it will report now no timestamp was found for that path. Which was better than the unclear error before.
- Added more docstrings to functions

A potential downside to this PR is that aside of the LakeFS instance connection it now uses _no_ local cache - which of course can be implemented in a separate PR later.

## Testing notes:

1. Clone the repository, checkout the branch (and don't forget all the git submodules, etc.)
    - ~~Note that there's a currently a non-working git submodule set (like in `develop` branch) for `ayon_bin_client`. Use branch `release_v0_1_0` instead for that submodule.~~ Fixed with https://github.com/ynput/ayon-usd/pull/36
2. Configure the settings using the LakeFS urls, etc. (request them with Lyon from Ynput team)
3. Launch Maya or Houdini
